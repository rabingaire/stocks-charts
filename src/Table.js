import React from "react";

function Table({ children }) {
  return (
    <table>
      <tbody>{children}</tbody>
    </table>
  );
}

function Row({ render }) {
  return render();
}

export { Table, Row };
