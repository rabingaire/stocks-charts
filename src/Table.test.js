// Link.react.test.js
import React from "react";
import { Table, Row } from "./Table";
import renderer from "react-test-renderer";

test("Table Gets Rendered", () => {
  const companies = [
    {
      name: "Apple",
      stockSymbol: "AAPL"
    },
    {
      name: "Facebook",
      stockSymbol: "FB"
    }
  ];

  const component = renderer.create(
    <Table>
      {companies.map(company => (
        <Row
          key={company.stockSymbol}
          render={() => (
            <tr>
              <td>{company.name}</td>
              <td>{company.stockSymbol}</td>
            </tr>
          )}
        />
      ))}
    </Table>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
