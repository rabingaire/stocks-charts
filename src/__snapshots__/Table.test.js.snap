// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Gets Rendered 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        Apple
      </td>
      <td>
        AAPL
      </td>
    </tr>
    <tr>
      <td>
        Facebook
      </td>
      <td>
        FB
      </td>
    </tr>
  </tbody>
</table>
`;
